### PR TITLE
Expand docs for `nn.functional`, and make the wording consistent

### LINKED ISSUE #148353: `torch.nn.functional` inconsistent documentation
### 📚 The doc issue

Functional versions of losses have inconsistent documentation, for example `torch.nn.functional.huber_loss` is well documented, including all parameters, where as `torch.nn.functional.l1_loss` has almost no documentation and is missing the `weight` parameter in the documentation, which is present in the code.

### Suggest a potential alternative/fix

Updating the documentation.

### PR DESCRIPTION
**Pull request by olipinski**
Expands the docs for the loss functions, and makes the wording consistent.

Fixes #148353

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thanks for take the time to go over this!

**Comment by olipinski** [comment]
Hi @albanD, it seems the merge won't go through without the labels, but I don't think I can add them myself, so I have requested re-review.

**Review by albanD — APPROVED** [review]
